src/firefly_exchange_client.py

```diff
@@ -297,7 +298,7 @@ def post_signed_order(self, params:PlaceOrderRequest):
             "price": to_big_number(params["price"]),
             "quantity": to_big_number(params["quantity"]),
             "leverage": to_big_number(params["leverage"]),
-            "userAddress": self.account.address.lower(),
+            "userAddress": params["maker"],
```

Comment:
is if "maker" in params check not required here

Reply:
The check is not needed here because `post_signed_order` takes `params:PlaceOrderRequest` as input and I have updated the interface of `PlaceOrderRequest` to have maker


```
class RequiredOrderFields(TypedDict): 
  symbol: MARKET_SYMBOLS # market for which to create order
  price: int # price at which to place order. Will be zero for a market order
  quantity: int # quantity/size of order
  side: ORDER_SIDE # BUY/SELL
  orderType: ORDER_TYPE # MARKET/LIMIT
  
class OrderSignatureResponse(RequiredOrderFields):
  maker: str 
  orderSignature: str

class PlaceOrderRequest(OrderSignatureResponse):
  timeInForce: TIME_IN_FORCE # FOK/IOC/GTT by default all orders are GTT
  postOnly: bool # true/false, default is true
  clientId: str # id of the client
  ```